Box.V2/Managers/BoxResourceManager.cs

```diff
@@ -257,6 +257,36 @@ protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQuer
             return allItemsCollection;
         }
 
+        /// <summary>
+        /// Used to fetch all results using pagination for metadata queries. V2 is for when a BoxItem is returned when fields are passed in.
+        /// </summary>
+        /// <typeparam name="T">The type of BoxCollectionMarkerBased item to expect.</typeparam>
+        /// <param name="request">The pre-configured BoxRequest object.</param>
+        /// <returns></returns>
+        protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQueryV2<T>(BoxRequest request) where T : BoxItem, new()
+        {
+            var allItemsCollection = new BoxCollectionMarkerBased<T>();
+            allItemsCollection.Entries = new List<T>();
+
+            bool keepGoing;
+            do
+            {
+                var response = await ToResponseAsync<BoxCollectionMarkerBased<T>>(request).ConfigureAwait(false);
+                var newItems = response.ResponseObject;
+                allItemsCollection.Entries.AddRange(newItems.Entries);
+                allItemsCollection.Order = newItems.Order;
+
+                dynamic body = JObject.Parse(request.Payload);
+                body.marker = newItems.NextMarker;
+                request.Payload = body.ToString();
+
+                keepGoing = !string.IsNullOrWhiteSpace(newItems.NextMarker);
+
+            } while (keepGoing);
+
+            return allItemsCollection;
```

Comment:
This is duplicated code with the method above.  Can this be refactored (extracted into a new method) so the duplicate code only lives in one place?

Reply:
I was thinking about this, but one of these methods is going to be removed once metadata query changes to support only one. Keeping it separate will make it easy to remove.